embedded-hal-async/src/spi.rs

```diff
@@ -35,13 +35,117 @@ where
     Word: Copy + 'static,
 = impl Future<Output = Result<(), T::Error>>;
 
+#[macro_export]
+/// This macro is a workaround for the workaround in [`SpiDevice::transaction`]
```

Comment:
Perhaps a more descriptive docs on why you might want to use it? Something like "Do an SPI transaction on a bus. This is a safe wrapper for [`SpiDevice::transaction`], which handles dereferencing the raw pointer for you."